src/main/java/frc/robot/commands/autonomousCommands/AutonomousShooterCommand.java

```diff
@@ -0,0 +1,81 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands.autonomousCommands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.HopperController;
+import frc.robot.subsystems.IndexerController;
+import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.shooter.ShooterSpeedController;
+import frc.robot.subsystems.shooter.ShooterMath;
+
+/**
+ * Add your docs here.
+ */
+public class AutonomousShooterCommand extends CommandBase{
+
+    private Drive drive;
+    private ShooterSpeedController shooterSpeedController;
+    private HopperController hopperController;
+    private IndexerController indexerController;
+    //variables
+    private double ballsShot = 0;
+    private double distanceToGoal = 149.0; 
+    //constants
+    private final double INNER_DISTANCE_FROM_TARGET = 29.0;
+
+    public AutonomousShooterCommand(Drive drive, ShooterSpeedController shooterSpeedController,
+    HopperController hopperController, IndexerController indexerController){
+        this.drive = drive;
+        this.shooterSpeedController = shooterSpeedController;
+        this.hopperController = hopperController;
+        this.indexerController = indexerController;
+
+        addRequirements(drive);
+    }
+
+    @Override
+    public void initialize(){
+
+    }
+
+    @Override
+    public void execute(){
+        shooterSpeedController.setLaunchSpeed(ShooterMath.determineLaunchSpeed(distanceToGoal + INNER_DISTANCE_FROM_TARGET));
+
+        for(int ballsShot = 0 ; ballsShot < 3 ; ballsShot ++){
+         // Make sure the shooter is at speed before loading a power cell
```

Comment:
This will increment ballsShot even if no power cell is launched.  You can leave the ifs the same, but instead of the for loop, whenever a ball is actually launched (in the first if), increment ballsShot.  We might want to add a timer to say run the indexer and hopper for 0.5 seconds each time.

Reply:
Example:
Add class members
private double startTime = -1.0;
private int ballsShot = 0;
private final double LAUNCH_TIME = 0.5;


         if (shooterSpeedController.isAtSpeed() && startTime < 0.0) {
            // turn on the indexer and hopper when at speed
            this.indexerController.intake();
            this.hopperController.start();
            startTime = Timer.getFPGATimestamp();
        }
        else if(shooterSpeedController.isAtSpeed() && (Timer.getFPGATimestamp() - startTime) < LAUNCH_TIME)
        {
              //keep the hopper running for LAUNCH_TIME to fire one ball
        }
        else if(shooterSpeedController.isAtSpeed() && (Timer.getFPGATimestamp() - startTime) > LAUNCH_TIME)
        {
            //after LAUNCH_TIME, stop the indexer/hopper and assume a ball launched.
            ++ballsShot;
             startTime = -1.0;
            indexerController.stop();
            hopperController.stop();
        }
        else {
            // turn off the indexer and hopper if not at speed and no ball launching
            indexerController.stop();
            hopperController.stop();
        }